### PR TITLE
Widget UI updates, fixing ipywidgets backward incompatible changes, config for adding endpoints and overriding auth

### PR DESCRIPTION
**Pull request by pranayhasan**
* Adding support for Cookies
* Upgraded ipywidgets to use Password Widget
* Setting password blank once user is authenticated
* Update widget width
* Adding authentication related properties. It will read these properties from config.json
* UI enhancements
* Changes to configure Auth Type
* Adding endpoints as a configuration
* Addresses #358

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ python:
 - '3.3'
 install:
 - pip install six
+- pip install --upgrade pip
```
Comment (aggFTW) [inline .travis.yml]: Why?

File: autovizwidget/setup.py [hunk 2]
```diff
@@ -58,8 +58,8 @@ def version(path):
         'Programming Language :: Python :: 3.4'],
       install_requires=[
           'plotly>=1.10.0,<1.11.0',
-          'ipywidgets>5.0.0,<7.0',
+          'ipywidgets>=7.0.0a4,<8.0',
```
Comment (aggFTW) [inline autovizwidget/setup.py]: Sorry, but let's not bump to any alpha version whatsoever. It should be `>=6.1.0,<7.0`

File: sparkmagic.iml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="PYTHON_MODULE" version="4">
```
Comment (aggFTW) [inline sparkmagic.iml]: Please remove all idea files from the review.

**Comment by pranayhasan** [comment]
@aggFTW Thanks! I will break them up and let you know.

**Comment by aggFTW** [comment]
Thanks so much! I will close this review then.

**Pull request closed without merging**